### PR TITLE
feat: Configurable backpack open

### PR DESCRIPTION
**Pull request by Chappie74**
**Description**

Added functionality for configuring whether the backpack should open if empty. 

**Testing**

When instantiating the backpack, add  `allowEmptyBackpackOpen: false` as an option. This defaults to true, which keeps the default blockly behaviour.

### CODE REVIEW

File: plugins/workspace-backpack/src/options.js [hunk 1]
```diff
@@ -44,6 +45,7 @@ export function parseOptions(options) {
       copyAllToBackpack: false,
       pasteAllToBackpack: false,
       disablePreconditionChecks: false,
+      allowEmptyBackpackOpen: true,
```
Comment (maribethb) [inline plugins/workspace-backpack/src/options.js]: This should be a new top-level option instead of being nested under contextMenu options. The context menu options configures which actions are available in the context menu, and this option is unrelated.

File: plugins/nominal-connection-checker/src/type_hierarchy.js [hunk 2]
```diff
@@ -497,16 +497,17 @@ export class TypeHierarchy {
...
+   * the paramters of to the common type.
    * @param {string} commonType The name of the common type we are getting the
-   *     params in the order of.
+   * params in the order of.
+   * @param getParamsForCommon
    * @param {function(!TypeDef, string, !Array<!TypeStructure>)
-   * :!Array<!TypeStructure>}
-   *     getParamsForCommon A function that takes in the definition of a type
-   *     and maps the given actual params to the parameters of the common type.
+:!Array<!TypeStructure>}
```
Comment (maribethb) [inline plugins/nominal-connection-checker/src/type_hierarchy.js]: Could you revert all the changes that aren't in the backpack plugin? They aren't related and some of the comments are incorrectly formatted - e.g. not starting the comment line with the asterisk can cause problems with jsdoc parsing.

File: plugins/workspace-backpack/src/backpack.js [hunk 3]
```diff
@@ -578,7 +578,8 @@ export class Backpack extends Blockly.DragTarget {
    * @protected
    */
   isOpenable_() {
-    return !this.isOpen();
+    return !this.isOpen() &&
+    this.options_.contextMenu.allowEmptyBackpackOpen? true: this.getCount() > 0;
```
Comment (maribethb) [inline plugins/workspace-backpack/src/backpack.js]: Surprised the linter didn't complain about this, but could you add a space before the `?` and `:` please: https://google.github.io/styleguide/jsguide.html#:~:text=On%20both%20sides%20of%20any%20binary%20or%20ternary%20operator.

and then this line will probably need to be wrapped to stay under 80 chars

**Review by maribethb — COMMENTED** [review]
Thank you for the PR and for contributing to Blockly! I think this just needs a bit of cleanup and then we should be good to go. Let me know if you have any questions!

Could you also add information about the new option in the README for this plugin?

**Comment by Chappie74** [comment]
Thanks for the feedback. Will take a look and fix the issues.

File: plugins/nominal-connection-checker/src/type_hierarchy.js (hunk 2 above)
Comment (Chappie74) [reply plugins/nominal-connection-checker/src/type_hierarchy.js]: Yeah I was wondering about all those changes. I couldn't get the project to run due to linting errors. And I ran the lint:fix command, which changed a lot of files.

**Review by alschmiedt — APPROVED** [review]

**Comment by alschmiedt** [comment]
Just looked it over and the changes look good to me.

Thank you for your contribution!
